src/GraphQL.Tests/ObjectExtensionsTest.cs

```diff
@@ -419,4 +432,208 @@ private class MyInput8
         public required string Name { get; set; } = "abc";
         public readonly int Age = -1;
     }
+
+    [Theory]
+    [InlineData(false, false)]
+    [InlineData(true, false)]
+    [InlineData(false, true)]
+    [InlineData(true, true)]
+    public void toobject_uses_valueconverter(bool compile, bool withValueConverter)
+    {
+        var inputs = """{ "name": "tom", "age": 10 }""".ToInputs();
+        if (withValueConverter)
+            ValueConverter.Register(_ => new MyInput9("testing"));
+        try
+        {
+            if (withValueConverter)
+            {
+                inputs.ToObject<MyInput9>(compile).ShouldNotBeNull();
+            }
+            else
+            {
+                Should.Throw<InvalidOperationException>(() => inputs.ToObject<MyInput9>(compile));
+            }
+        }
+        finally
+        {
+            ValueConverter.Register<MyInput9>(null);
+        }
+    }
+
+    private class MyInput9
+    {
+        public MyInput9(string dummy) { _ = dummy; }
+        public string Name { get; set; }
+        public int Age { get; set; }
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public void toobject_uses_constructor_with_optional_parameters(bool compile)
+    {
+        var inputs = """{ "name": "tom" }""".ToInputs();
+        var value = inputs.ToObject<MyInput10>(compile).ShouldNotBeNull();
+        value.Name.ShouldBe("tom");
+    }
+
+    public class MyInput10
+    {
+        public MyInput10(string name, int age = 10)
+        {
+            Name = name;
+            age.ShouldBe(10);
+        }
+
+        public string Name { get; }
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public void toobject_throws_for_constructor_with_unknown_parameters(bool compile)
+    {
+        var inputs = """{ "name": "tom" }""".ToInputs();
+        Should.Throw<InvalidOperationException>(() => inputs.ToObject<MyInput11>(compile))
+            .Message.ShouldBe("Cannot find field named 'age' on graph type 'MyInput11' to fulfill constructor parameter for CLR type 'MyInput11'.");
```

Comment:
Didn't you say you [want](https://github.com/graphql-dotnet/graphql-dotnet/issues/3757#issuecomment-1807297711) to coerce this to default value in this case?

Reply:
In this case `age` isn't defined on the graph type.  So, no, if it encounters an unrecognized constructor parameter without a default value, it throws.